cmd/juju/application/deployer/deployer.go

```diff
@@ -104,7 +104,11 @@ func (d *factory) GetDeployer(ctx context.Context, cfg DeployerConfig, getter Mo
 		return nil, errors.Trace(fileStatErr)
 	}
 
-	if charm.IsValidLocalCharmOrBundlePath(d.charmOrBundle) || isLocalSchema(d.charmOrBundle) {
+	if isLocalSchema(d.charmOrBundle) && charm.IsValidLocalCharmOrBundlePath(d.charmOrBundle[6:]) {
```

Comment:
This will never be true. I don't believe an OR is appropriate either based on what is done when true. 

What is the purpose here?

Reply:
This will be true when you run `juju deploy local:/path/to/charm`

This is a pretty nasty format, so the purpose of this `if` is to simplify down `charmOrBundle` down from `local:/path/to/charm` down to `/path/to/charm`

As you can see on the parent PR, I elected for a slightly different approach in the end